src/notation/view/percussionpanel/percussionpanelpadlistmodel.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "percussionpanelpadlistmodel.h"
+
+PercussionPanelPadListModel::PercussionPanelPadListModel(QObject* parent)
+    : QAbstractListModel(parent)
+{
+    m_padModels = defaultItems();
```

Comment:
It's better to add Q_INVOKABLE load() and call it from Component.onCompleted instead of filling the model inside the constructor